### PR TITLE
Add flake8 to automated linting and testing

### PR DESCRIPTION
**Pull request by necrobuffalo**

### CODE REVIEW

**Comment by J8cob1** [comment]
Remember to extend the line length: flake8 file.blah --max-line-length=110

File: gbmodel/model_sqlalchemy.py [hunk 1]
```diff
@@ -147,13 +148,14 @@ def remove_student(self, sts, t_name, session_id):
...
-        result = engine.execute("select id from teams where name = :name AND session_id = :session_id", params)
+        result = engine.execute("select id from teams where name = :name AND session_id = :session_id",
+                                params)
         id = result.fetchone()
         tid = id[0]
 
         for i in sts:
             params = {'name': i, 'tid': tid, 'session_id': session_id}
-            result = engine.execute("select id, tid, session_id, name, is_lead, midterm_done, final_done from students where name = :name AND tid= :tid AND session_id = :session_id", params)
+            result = engine.execute("select id, tid, session_id, name, is_lead, midterm_done, final_done from students where name = :name AND tid= :tid AND session_id = :session_id", params)  # noqa
```
Comment (J8cob1) [inline gbmodel/model_sqlalchemy.py:158]: Longer than 110 characters?

File: .circleci/config.yml [hunk 2]
```diff
@@ -21,10 +21,18 @@ jobs:
...
+      - run: sudo pip3 install -U flake8
+      - run: flake8 --max-line-length=110
 
 workflows:
   version: 2
   build_and_test:
     jobs:
       - test
+      - lint
       # - mypy
```
Comment (J8cob1) [inline .circleci/config.yml:38]: You know better than me here

File: gbmodel/model_sqlalchemy.py (hunk 1 above)
Comment (necrobuffalo) [reply gbmodel/model_sqlalchemy.py:158]: Since we're switching to the ORM anyway I'm leaving this in and silencing it for now, then will fix in a separate PR.

File: .circleci/config.yml (hunk 2 above)
Comment (necrobuffalo) [reply .circleci/config.yml:38]: Their docs are decent, if you're interested. https://circleci.com/docs/2.0/config-intro/#section=configuration

File: gbmodel/model_sqlalchemy.py (hunk 1 above)
Comment (necrobuffalo) [reply gbmodel/model_sqlalchemy.py:158]: (the `# noqa` bit at the end of the line makes flake8, mypy, and presumably other linters ignore the line)

**Review by pdx-cs-lbryan — APPROVED** [review]
Is your task complete?
Adds flake8 linting to circleci tests

Is the branch caught up to master/are there merge conflicts still to be resolved?
ahead of master

Are old files unnecessarily introduced?
No

Is old code unnecessarily overwritten?
No

Are new issues introduced to existing code? Do existing automated tests continue to work?
Log in: Works
Dashboard/Edit: Works
Review: Works
Student Forms: Works
Prof Forms: Works
New Issues: None
Automated Tests: pass

Has flake8 been checked?
yes

Uses 4 spaces where appropriate, not tabs?
yes

camelCase for class names, snake_case for variable and function names?
yes

code is commented?
not necessary

Does the code handle edge cases? Are try/except statements used sufficiently?
only restylizing

Is there documentation on how to test your code somewhere?
~
